apps/meteor/tests/end-to-end/api/livechat/18-rooms-ee.ts

```diff
@@ -254,4 +254,76 @@ import { IS_EE } from '../../../e2e/config/constants';
 			expect(updatedRoom).to.not.have.property('onHold');
 		});
 	});
+
+	describe('visitor abandonment feature', () => {
+		let room: IOmnichannelRoom;
+
+		before(async () => {
+			await updateSetting('Livechat_abandoned_rooms_action', 'Livechat_close_chat');
+			await updateSetting('Livechat_visitor_inactivity_timeout', 60);
+		});
+
+		it('should set predictedVisitorAbandonmentAt when agent sends a message', async () => {
+			const { room: newRoom } = await startANewLivechatRoomAndTakeIt();
+
+			room = newRoom;
+
+			await sendAgentMessage(room._id);
+
+			const updatedRoom = await getLivechatRoomInfo(room._id);
+
+			const lastMessageTs = updatedRoom.responseBy?.lastMessageTs;
+			const firstResponseTs = updatedRoom.responseBy?.firstResponseTs;
+			const predictedVisitorAbandonmentAt = updatedRoom.omnichannel?.predictedVisitorAbandonmentAt;
+
+			expect(predictedVisitorAbandonmentAt).to.not.be.undefined;
+			expect(lastMessageTs).to.not.be.undefined;
+			expect(firstResponseTs).to.not.be.undefined;
+
+			if (!lastMessageTs || !predictedVisitorAbandonmentAt || !firstResponseTs) {
```

Comment:
I believe the `expect`s statements above are already checking this, right? I think we can remove this validation.

Reply:
Yes you're right but typescript isn't as smart as you :trollface: ; as it was complaining about `predictedVisitorAbandonmentAt` being undefined when I was using it after expect checks. Hence I'd to add these if conditions to make it infer that it's not null now.

But in hindsight: I think I could've just used a typecast here